src/main/java/bern/functionalities/Parser.java

```diff
@@ -0,0 +1,165 @@
+import bern.model.Event;
+import bern.model.Task;
+import bern.model.ToDo;
+
+/**
+ * A class to contain all methods parsing related.
+ */
+public class Parser {
+    /**
+     * A method to create a ToDo instance from a string command.
+     *
+     * @param s The command from which a ToDo instance will be made.
+     * @return The ToDo instance.
+     */
+    public ToDo toDoFromString(String s) {
+        boolean isDone = s.substring(4, 5).equals("X");
+        String desc = s.substring(7);
+        ToDo ans = new ToDo(desc);
+        if (isDone) {
+            ans.markAsDone();
+        }
+        return ans;
+    }
+
+    /**
+     * A method to create a Deadline instance from a string command.
+     *
+     * @param s The command from which a Deadline instance will be made.
+     * @return The Deadline instance.
+     */
+    public Deadline deadlineFromString(String s) {
+        boolean isDone = s.substring(4, 5).equals("X");
+        String desc = s.substring(7, s.indexOf("(by: ") - 1);
+        String by = s.substring(s.indexOf("(by: ") + 5, s.length() - 1);
+        Deadline ans = new Deadline(desc, by);
+        if (isDone) {
+            ans.markAsDone();
+        }
+        return ans;
+    }
+
+    /**
+     * A method to create an Event instance from a string command.
+     *
+     * @param s The command from which an Event instance will be made.
+     * @return The Event instance.
+     */
+    public Event eventFromString(String s) {
+        boolean isDone = s.substring(4, 5).equals("X");
+        String desc = s.substring(7, s.indexOf("(at: ") - 1);
+        String at = s.substring(s.indexOf("(at: ") + 5, s.length() - 1);
+        Event ans = new Event(desc, at);
+        if (isDone) {
+            ans.markAsDone();
+        }
+        return ans;
+    }
+
+    /**
+     * A method to create a Task instance (can be Deadline, Event, or ToDo) from a string command.
+     *
+     * @param s The command from which a Task instance will be made.
+     * @return The Task instance.
+     */
+    public Task taskFromString(String s) {
+        String cat = s.substring(1, 2);
+        return cat.equals("T")
+                ? toDoFromString(s)
+                : cat.equals("D")
+                ? deadlineFromString(s)
+                : eventFromString(s);
+    }
+
+    /**
+     * A method to check whether a given string can be parsed as a number (integer).
+     *
+     * @param s The given string.
+     * @return A boolean to indicate whether it's a number or not.
+     */
+    public boolean isANumber(String s) {
+        try {
+            Integer.parseInt(s);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+
+    /**
+     * A method to check whether a command is of category "Done".
+     *
+     * @param input The command given.
+     * @return Boolean whether the command is of category "Done".
+     */
+    public boolean isDone(String input) {
```

Comment:
I feel like creating a method for each keyword might gradually get messy over time. maybe consider a generalised alternative?

```Java
// Return whether first word of input matches the given keyword
public static boolean isKeyword(String input, String keyword) {
    String firstWord = input.split(" ", 2)[0];  //Get first word in input
    return firstWord.equals(keyword);
}
```

Again, not really a coding style violation, but I thought this could be useful.